app/models/hyrax/collection_type.rb

```diff
@@ -22,8 +22,23 @@ class CollectionType < ActiveRecord::Base
     alias_attribute :visibility, :assigns_visibility
     deprecation_deprecate visibility: "prefer #assigns_visibility instead"
 
+    # Find the collection type associated with the Global Identifier (gid)
+    # @param [String] gid - Global Identifier for this collection_type (e.g. gid://internal/hyrax-collectiontype/3)
+    # @return [Hyrax::CollectionType] an instance of Hyrax::CollectionType with id = the model_id portion of the gid (e.g. 3), or nil if gid not found
+    def self.find_by_gid(gid)
+      find(GlobalID.new(gid).model_id)
+    rescue ActiveRecord::RecordNotFound
+      nil
+    end
+
+    # Return the Global Identifier for this collection type.
+    # @return [String] Global Identifier (gid) for this collection_type (e.g. gid://internal/hyrax-collectiontype/3)
     def gid
-      URI::GID.build app: GlobalID.app, model_name: model_name.name.parameterize.to_sym, model_id: id unless id.nil?
+      return nil if id.nil?
+      # TODO: From a modeling perspective, I'm not sure which of these we should make the <object_value>.  For now, we will store in Fedora object as string, but might want to revisit.
+      uri_gid = URI::GID.build(app: GlobalID.app, model_name: model_name.name.parameterize.to_sym, model_id: id) unless id.nil? # ActiveTriples won't accept this as an <object_value>
+      # rdf_uri = RDF::URI.new(uri_gid) # ActiveTriples converts this to an ActiveTriples relationship that is a new triple with <subject> = gid uri
```

Comment:
It is there associated with the TODO question to document how to generate an RDF::URI version of the gid and what ActiveTriples does with it.

The way the code is written right now, the gid gets stored in Fedora as a string with format `'gid://internal/hyrax-collectiontype/:id'`.  I'm not sure that is what we want from a modeling perspective.  But ActiveTriples doesn't seem too happy with it.  So I'm punting it to later. 

I'll move it up one line, closer to the TODO and add a TODO marker to it.